designs/virtualization/4.6/Add-os-image-to-common-template/add-os-image-to-common-template 4.6.md

```diff
@@ -0,0 +1,56 @@
+---
+parent: Virtualization
+version: 4.6
+---
+# Add an operating system image to a common template
+
+## About
+
+We strive to provide an off-the-shelf OS image with every common template.
+This an ongoing effort.
+Hopefully this will provide the user with a much easier experience to creating a virtual machine.
+This OS image takes the shape of a disk image(data-volume) and can be attached to the OS template (one of the parameters within the common templates).
+These OS images, and common templates in general, won’t be editable for the user.
+Today, we don’t have OS images, so we need a temporary solution; offering the user a way to attach a new image that he adds to a common template, making it available on selecting that OS template field.
+
+
+## Adding an OS image
+
+![Create VM wizard - default](img/Admin-create-4.6-0-0.jpg)
+
+On clicking "Create VM" the user lands on the wizard.
+The user enters a VM name and clicks on the operating system dropdown to select an OS.
+
+![OS dropdown](img/Admin-create-4.6-0-1.jpg)
+
+The OS dropdown would promote OSs backed by images by placing them on top.
+It will also have a search field to fetch all desired results without going through two lists.
+This time, the user selects an OS without an available image/source.
+
+![No OS image available warning](img/Admin-create-4.6-0-2.jpg)
+
+The user will be presented with a warning message notifying that no OS image/source detected.
+The user can either click the "Upload and attach..." action to couple a new OS image to the OS selection, or to provide a source on the field below.
+A “source” field (previously called “source”) will present itself.
+
+!["Create VM session will be discarded"](img/Admin-create-4.6-0-3.jpg)
```

Comment:
I think the Alert convention would have this inline alert at the top of the modal, since it's there when it is opened.

Reply:
nice catch @itsptk

Reply:
Not sure I follow, you mean to place this notification as an inline alert within the wizard page?

If so, I don't think I agree since:

- It is a destructive action, so it is better to block the user with a modal to confirm that to mention it next to a button where it might be missed.
- It has a lot of into going on the modal. Adding this info as an inline alert to what is an optional path in our (currently) main flow of creating a VM is simply white noise, for most of the time.

Reply:
Sorry, I should have just linked to the convention initially. I was just trying to say that I think we decided if a modal opens with an inline alert already in place, it should appear at the top of the modal: http://openshift.github.io/openshift-origin-design/conventions/documentation/alerting.html#at-top-of-modal